rx-preferences/src/main/java/com/f2prateek/rx/preferences2/CustomTypeStringAdapter.java

```diff
@@ -0,0 +1,29 @@
+package com.f2prateek.rx.preferences2;
+
+import android.content.SharedPreferences;
+import android.support.annotation.NonNull;
+
+import static com.f2prateek.rx.preferences2.Preconditions.checkNotNull;
+
+final class CustomTypeStringAdapter<T> implements RealPreference.Adapter<T> {
```

Comment:
I think the name `CustomTypeStringAdapter` could be improved. Possibly something involving `Object` and/or `Converter` — e.g. `ObjectAdapter`, `ConverterAdapter`, `ObjectConverterAdapter`?